registry/types.go

```diff
@@ -1,26 +1,43 @@
 package registry
 
+// SearchResult describes a search result returned from a registry
 type SearchResult struct {
-	StarCount   int    `json:"star_count"`
-	IsOfficial  bool   `json:"is_official"`
-	Name        string `json:"name"`
-	IsTrusted   bool   `json:"is_trusted"`
-	IsAutomated bool   `json:"is_automated"`
+	// StarCount indicates the number of stars this repository has
+	StarCount int `json:"star_count"`
+	// IsOfficial indicates whether the result is an official repository or not
+	IsOfficial bool `json:"is_official"`
+	// Name is the name of the repository
+	Name string `json:"name"`
+	// IsOfficial indicates whether the result is trusted
+	IsTrusted bool `json:"is_trusted"`
+	// IsAutomated indicates whether the result is automated
+	IsAutomated bool `json:"is_automated"`
+	// Description is a textual description of the repository
 	Description string `json:"description"`
 }
 
+// SearchResults lists a collection search results returned from a registry
 type SearchResults struct {
-	Query      string         `json:"query"`
-	NumResults int            `json:"num_results"`
-	Results    []SearchResult `json:"results"`
+	// Query contains the query string that generated the search results
+	Query string `json:"query"`
+	// NumResults indicates the number of results the query returned
+	NumResults int `json:"num_results"`
+	// Results is a slice containing the acutal results for the search
+	Results []SearchResult `json:"results"`
 }
 
+// RepositoryData tracks the image list, list of endpoints, and list of tokens
+// for a repository
 type RepositoryData struct {
-	ImgList   map[string]*ImgData
+	// ImgList is a list of images in the repository
+	ImgList map[string]*ImgData
+	// Endpoints is a list of endpoints returned in X-Docker-Endpoints
 	Endpoints []string
-	Tokens    []string
+	// Tokens is currently unused (remove it?)
+	Tokens []string
 }
 
+// ImgData describes an image
```

Comment:
This type is used by the v1 code to store the list of images which will be pushed or pulled. My understanding is the "Checksum" in this struct has no significant use.